### DISCUSSION TITLE
Any chance of adding Gemini and Copilot to the program?

### DISCUSSION THREAD

**Discussion by AdrianeNS** [danny-avila/LibreChat#2703] [Feature Requests & Suggestions]
It would be amazing to be able to switch between the big 3!  Since you already have ChatGPT, what would it take to add both Google's Gemini and Microsoft's Copilot?  Thanks for your consideration.

**Comment by danny-avila** [comment]
Hi Adriane, thanks for your post.

There's no official API access for Copilot, and Gemini is already added. Copilot is using the OpenAI suite of models under the hood, anyway.

**Comment by frank690** [comment]
hey danny. shortly after your response, API access to copilot was actually provided by microsoft. even though it uses openai under the hood, many people (and businesses) have a paid plan that includes the use of copilot. imo it would be great to use librechat and integrate copilot in here. or am i missing something? thanks in advance!

**Reply by danny-avila** [reply]
If Copilot exposes an OpenAI-compatible endpoint, they would effectively make it accessible to many different apps, and it could be added as a custom endpoint for LibreChat:

https://www.librechat.ai/docs/quick_start/custom_endpoints

**Comment by Ancientkingg** [comment]
With the recent addition of the copilot-sdk (https://github.com/github/copilot-sdk). Would there be any possibility of adding copilot to librechat?

**Reply by Shentoza** [reply]
However that seems to be the "Github Copilot" SDK, which is not to be confused with "Microsoft 365 Copilot"
